scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -1966,11 +1999,12 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
       // simple diagnostics for this entry point
       simplePattern(token => syntaxError("illegal start of simple pattern", at = token))
     def simplePattern(onError: Token => Nothing): Pat = autoPos(token match {
-      case _: Ident | _: `this` =>
-        val isBackquoted = token.code.endsWith("`")
+      case _: Ident | _: `this` | _: Unquote =>
+        val isBackquoted = parser.isBackquoted
         val sid = stableId()
         val isVarPattern = sid match {
-          case Term.Name(value) => !isBackquoted && value.head.isLower && value.head.isLetter
+          case _: Term.Name.Quasi => false
+          case Term.Name(value) => !isBackquoted && value.head.isLetter && value.head.isLower
```

Comment:
`Term.Name.Quasi` is `Term.Name`, but trying to extract `value` from first one leads to error.